accounts/ethstore/src/account/safe_account.rs

```diff
@@ -77,16 +77,32 @@ impl SafeAccount {
 	/// Create a new `SafeAccount` from the given `json`; if it was read from a
 	/// file, the `filename` should be `Some` name. If it is as yet anonymous, then it
 	/// can be left `None`.
-	pub fn from_file(json: json::KeyFile, filename: Option<String>) -> Self {
-		SafeAccount {
+	pub fn from_file(json: json::KeyFile, filename: Option<String>) -> Result<Self, Error> {
```

Comment:
This now errors for no good reason, afaict it will never fail. Can we please get the proper return type back?

Reply:
it will fail for example when the password is incorrect.